fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java

```diff
@@ -405,28 +165,45 @@ private void handleSocialMediaNavigation() {
     if (mViewModel.getLAOs().getValue() == null) {
       ErrorUtils.logAndShow(
           getApplicationContext(), TAG, new NoLAOException(), R.string.error_no_lao);
-      revertToHome();
+      showHomeTab();
     } else {
-      mViewModel.openSocialMedia();
+      HomeViewModel.openSocialMedia(this);
     }
   }
 
   private void handleConnectNavigation() {
     if (checkWalletInitialization()) {
-      mViewModel.openConnect();
+      openConnect();
     } else {
-      revertToHome();
+      showHomeTab();
     }
   }
 
   private void handleLaunchNavigation() {
     if (checkWalletInitialization()) {
-      mViewModel.openLaunch();
+      mViewModel.openLaunch(getSupportFragmentManager());
+    } else {
+      showHomeTab();
+    }
+  }
+
+  public void openConnect() {
+    if (checkSelfPermission(Manifest.permission.CAMERA) == PackageManager.PERMISSION_GRANTED) {
+      mViewModel.openQrCodeScanning(getSupportFragmentManager());
     } else {
-      revertToHome();
+      requestCameraPermission();
     }
   }
 
+  public void requestCameraPermission() {
+    // Setup result listener to open the connect tab once the permission is granted
+    getSupportFragmentManager()
+        .setFragmentResultListener(
+            CameraPermissionFragment.REQUEST_KEY, this, (k, b) -> openConnect());
+
+    mViewModel.openCameraPermission(getSupportFragmentManager(), getActivityResultRegistry());
+  }
```

Comment:
what happens if the camera permission is denied ?
I don't see a code path to deal with that situation / log it / warn the user in some way.

Reply:
It will reopen Connect which will redo the permission check and if the test fails it will re-ask the user.

It is basically a loop.

Reply:
Upon testing, I noticed that when denying, we go to the home fragment. In itself, it is not a problem but after twice denying, when clicking on the allow button (ours) it does not ask for permission but goes back to home like I denied permission

Moreover, I think the behaviour should be commented and if possible the logic behind. Imo it is not immediately clear from the code.